packages/babel-helper-builder-react-jsx-experimental/src/index.js

```diff
@@ -804,8 +833,35 @@ You can set \`throwIfNamespace: false\` to bypass this warning.`,
    * breaking on spreads, we then push a new object containing
    * all prior attributes to an array for later processing.
    */
-  function buildCreateElementOpeningElementAttributes(path, attribs) {
-    const props = [];
+  function buildCreateElementOpeningElementAttributes(file, path, attribs) {
+    // TODO (Babel 8): Only leave this branch of the code and remove the rest
+    if (
+      RUNTIME_DEFAULT === "automatic" ||
```

Comment:
Did you mean `"classic"` here? IIRC we decided we always use spread in these cases for `"automatic"` runtime.

Reply:
And this branch contains only the logic that uses spread. If either the option or pragma configures the runtime to the automatic one we stick to that.

Maybe this looks a little bit weird because this branch contains the final logic while looking like a special case to be removed. I've chosen this as it's way shorter than what we need to support right now the the classic runtime so it made sense to me to put the shorter block indented.

Reply:
Also, note that this block literally contains the code that was already here in this function before my changes - and it was intended only to handle the automatic runtime.